src/appleseed/renderer/modeling/surfaceshader/diagnosticsurfaceshader.cpp

```diff
@@ -576,11 +580,23 @@ void DiagnosticSurfaceShader::evaluate(
         }
         break;
 
+      case Assemblies:
+        set_shading_result(
+            shading_result,
+            integer_to_color3<float>(shading_point.get_assembly_instance().find_assembly()->get_uid()));
```

Comment:
Use `ShadingPoint::get_assembly()` which is very fast. `AssemblyInstance::find_assembly()` is very slow, it's used to find the assembly outside of rendering.